packages/shared/styles/components/atoms/SfCheckbox.scss

```diff
@@ -7,51 +7,62 @@
     width: var(--checkbox-container-width);
     cursor: var(--checkbox-cursor, pointer);
   }
+
   &__checkmark {
     box-sizing: border-box;
     display: flex;
     align-items: center;
     justify-content: center;
-    width: var(--checkbox-width, var(--checkbox-size, 1.25rem));
-    height: var(--checkbox-height, var(--checkbox-size, 1.25rem));
-    border: var(--checkbox-border, 1px solid var(--c-gray-variant));
+    width: var(--checkbox-width, var(--checkbox-size, 1.5rem));
+    height: var(--checkbox-height, var(--checkbox-size, 1.5rem));
+    @include border(--checkbox-border, 1px, solid, var(--c-gray-variant));
     border-radius: var(--checkbox-border-radius);
     background: var(--checkbox-background, transparent);
     transition: background 150ms linear, border-color 150ms linear;
     &:hover {
-      --checkbox-border: 1px solid var(--c-primary);
+      --checkbox-border-color: var(--c-primary);
       .sf-checkbox--is-disabled & {
-        --checkbox-border: 1px solid var(--c-text-muted);
+        --checkbox-border-color: var(--c-text-muted);
+      }
+      .sf-checkbox--has-error & {
+        --checkbox-border-color: var(--c-danger);
       }
     }
   }
+
   &__label {
     flex: 1;
-    margin: var(--checkbox-label-margin, 0 0 0 var(--spacer-big));
+    margin: var(--checkbox-label-margin, 0 0 0 var(--spacer-small));
     color: var(--checkbox-label-color, var(--c-text));
-    font: var(
+    @include font(
       --checkbox-font,
-      font(
-        --checkbox,
-        300,
-        var(--font-size-big),
-        1.6,
-        var(--body-font-family-secondary)
-      )
+      500,
+      var(--font-size-extra-small),
+      1.6,
+      var(--body-font-family-secondary)
     );
   }
+
   &--is-active {
-    --checkbox-border: 1px solid var(--c-primary);
+    --checkbox-border-color: var(--c-primary);
     --checkbox-background: var(--c-primary);
+
     &.sf-checkbox--is-disabled {
-      --checkbox-border: 1px solid var(--c-text-muted);
+      --checkbox-border-color: var(--c-text-muted);
       --checkbox-background: var(--c-text-muted);
     }
+
+    &.sf-checkbox--has-error {
+      --checkbox-border-color: var(--c-danger);
+      --checkbox-background: var(--c-danger);
+    }
   }
+
   &--is-disabled {
     --checkbox-label-color: var(--c-text-muted);
-    --checkbox-border: 1px solid var(--c-text-muted);
+    --checkbox-border-color: var(--c-text-muted);
```

Comment:
@martaradziszewska we have now `--c-text-disabled` we can use it for text and border-color wgyt?